File: SimpleImageViewer/SimpleImageViewerJVM/src/main/java/SimpleImageViewerJVM/FullImageViewer.java
package SimpleImageViewerJVM;

import java.awt.Color;

import com.spvessel.spacevil.*;
import com.spvessel.spacevil.Decorations.CustomFigure;
import com.spvessel.spacevil.Flags.*;

public class FullImageViewer extends DialogItem {
    private ButtonCore _close = new ButtonCore();
    private ImageItem _image;

    public FullImageViewer(ImageItem image) {
        _image = image;
    }
    
    @Override
    public void initElements() {
        super.initElements();
        setBackground(0, 0, 0, 200);

        _image.keepAspectRatio(true);

        _close.setSize(30, 30);
        _close.setCustomFigure(new CustomFigure(false, GraphicsMathService.getCross(30, 30, 3, 45)));
        _close.setBackground(100, 100, 100);
        _close.setAlignment(ItemAlignment.TOP, ItemAlignment.RIGHT);
        _close.setMargin(0, 10, 10, 0);

        window.setSizePolicy(SizePolicy.EXPAND, SizePolicy.EXPAND);
        window.setMargin(20, 20, 20, 20);
        window.setBackground(new Color(0, 0, 0, 0));
        addItems(_close);

        window.addItem(_image);

        _close.eventMouseClick.add((sender, args) -> {
            close();
        });
    }

    @Override
    public void show(CoreWindow handler) {
        super.show(handler);
    }

    @Override
    public void close() {
        if (onCloseDialog != null)
            onCloseDialog.execute();

        super.close();
    }
}

File: SimpleImageViewer/SimpleImageViewerNet/FullImageViewer.cs
using System.Drawing;
using SpaceVIL.Core;
using SpaceVIL;
using SpaceVIL.Decorations;

namespace SimpleImageViewer
{
    public class FullImageViewer : DialogItem
    {
        private ButtonCore _close = new ButtonCore();
        private ImageItem _image;
        public FullImageViewer(ImageItem image)
        {
            _image = image;
        }

        public override void InitElements()
        {
            base.InitElements();
            SetBackground(0, 0, 0, 200);
            
            Window.IsXResizable = false;
            Window.IsYResizable = false;
            Window.IsXFloating = false;
            Window.IsYFloating = false;

            _image.KeepAspectRatio(true);

            _close.SetSize(30, 30);
            _close.SetCustomFigure(new Figure(false, GraphicsMathService.GetCross(30, 30, 3, 45)));
            _close.SetBackground(100, 100, 100);
            _close.SetAlignment(ItemAlignment.Top, ItemAlignment.Right);
            _close.SetMargin(0, 10, 10, 0);

            Window.SetSizePolicy(SizePolicy.Expand, SizePolicy.Expand);
            Window.SetMargin(20, 20, 20, 20);
            Window.SetBackground(Color.Transparent);
            AddItems(_close);

            Window.AddItem(_image);

            _close.EventMouseClick += (sender, args) =>
            {
                Close();
            };
        }

        public override void Show(CoreWindow handler)
        {
            base.Show(handler);
        }

        public override void Close()
        {
            if (OnCloseDialog != null)
                OnCloseDialog.Invoke();

            base.Close();
        }
    }
}
